patroni/postgresql/__init__.py

```diff
@@ -711,9 +718,42 @@ def _wait_promote(self, wait_seconds):
             if data.get('Database cluster state') == 'in production':
                 return True
 
+    def _call_pre_promote(self, cmd):
+        if cmd:
+            try:
+                ret = self.cancellable.call(shlex.split(cmd))
+            except OSError as err:
```

Comment:
The `cancellable.call()` may raise the `PostgresException`...
In order to be safe, it is better to catch all `Exception` here.